var/spack/repos/builtin/packages/glib/package.py

```diff
@@ -118,6 +119,12 @@ class Glib(Package):
         description="Enable tracing support",
     )
 
+    build_system(
+        conditional("meson", when="@2.58:"),
+        conditional("autotools", when="@:2.57"),
+        default="meson",
+    )
+
     depends_on("meson@0.49.2:", when="@2.61.2:", type="build")
     depends_on("meson@0.48.0:", when="@2.58.0:", type="build")
     depends_on("ninja", when="@2.58.0:", type="build")
```

Comment:
Maybe wrap those in

```python
with when("build_system=meson"):
  ...
```

and drop the `@2.58.0:` lowerbound

Reply:
Good point, done! I've also removed the ninja dependency added implicitly by `MesonPackage` and updated the required meson versions for the later releases.